Orange/widgets/visualize/widgetutils/tree/skltreeadapter.py

```diff
@@ -0,0 +1,302 @@
+from collections import OrderedDict
+from functools import lru_cache
+
+import numpy as np
+from Orange.widgets.visualize.widgetutils.tree.treeadapter import TreeAdapter
+
+from Orange.preprocess.transformation import Indicator
+from Orange.widgets.visualize.widgetutils.tree.rules import (
+    DiscreteRule,
+    ContinuousRule
+)
+
+
+class SklTreeAdapter(TreeAdapter):
+    """Sklear Tree Adapter.
+
+    An abstraction on top of the scikit learn classification tree.
+
+    Parameters
+    ----------
+    tree : sklearn.tree._tree.Tree
+        The raw sklearn classification tree.
+    domain : Orange.base.domain
+        The Orange domain that comes with the model.
+    adjust_weight : Callable, optional
+        If you want finer control over the weights of individual nodes you can
+        pass in a function that takes the existsing weight and modifies it.
+        The given function must have signture :: Number -> Number
+
+    """
+
+    ROOT_PARENT = -1
+    NO_CHILD = -1
+    FEATURE_UNDEFINED = -2
+
+    def __init__(self, tree, domain, adjust_weight=lambda x: x):
+        self._tree = tree
+        self._domain = domain
+        self._adjust_weight = adjust_weight
+
+        # clear memoized functions
+        self.weight.cache_clear()
+        self._adjusted_child_weight.cache_clear()
+        self.parent.cache_clear()
+
+        self._all_leaves = None
+
+    @lru_cache(maxsize=1024)
```

Comment:
`lru_cache`-ing of methods may prevent disposing of self. https://stackoverflow.com/questions/33672412/python-functools-lru-cache-with-class-methods-release-object